addons/survey/static/src/js/survey_session_chart.js

```diff
@@ -166,6 +166,66 @@ publicWidget.registry.SurveySessionChart = publicWidget.Widget.extend({
                     }
                 }
             },
+            plugins: [{
+                /**
+                 * Adapt the labels of the graph to avoid overlapping text.
```

Comment:
I would elaborate a bit on what it does.
Something like:

```js
/** 
* The way it works is each label of the chart legend is an array of words.
*  eg.: If we have a chart label: "this is an example of a label"
* The library will split it as: ["this is an example", "of a label"]
* Each value of the array represents a line of the label
* So for this example above: it will be displayed as:
* "this is an example<br/>of a label", breaking the label in 2 parts and put it on 2 lines visually
* 
* What we do here is rework the labels with our own algorithm to make them fit better in screen space
* based on breakpoints based on number of columns to display.
* So this example will become: ["this is an", "example of", "a label"] if we have a lot of labels to put in the chart.
* Which will be displayed as "this is an<br/>example of<br/>a label"
* Obviously, the more labels you have, the more columns, and the less screen space is available.
* 
* We also adapt the font size based on the width available within the chart.
* 
* So we counterbalance twice:
* - Based on number of columns (i.e number of survey.question.answer of your current survey.question)
*   We split the words of every label to make them display on more rows
* - Based on the width of the chart (which is equivalent to screen width)
*   We reduce the chart font to be able to fit more characters
*/
```

If anything is incorrect, feel free to update above comment.
Maybe I did not understand something correctly.